In [27]:
report_card_dictionary={"Student_names":["Vanshika","Vivaan","Vrutik","kareena","Mitansh"],
                       "chemistry grades":[23,45,45,67,89],
                       "physics_grades":[23,56,78,98,67],
                       "math_grades":[56,76,87,54,98],
                       "biology_grades":[45,78,98,56,45]}

report_card_dictionary

{'Student_names': ['Vanshika', 'Vivaan', 'Vrutik', 'kareena', 'Mitansh'],
 'chemistry grades': [23, 45, 45, 67, 89],
 'physics_grades': [23, 56, 78, 98, 67],
 'math_grades': [56, 76, 87, 54, 98],
 'biology_grades': [45, 78, 98, 56, 45]}

In [28]:
# Setup
import pandas as pd

In [29]:
# Making the report card dictionary into a data frame
report_card_df=pd.DataFrame(report_card_dictionary)
report_card_df

Student_names  chemistry grades  physics_grades  math_grades  biology_grades
0      Vanshika                23              23           56              45
1        Vivaan                45              56           76              78
2        Vrutik                45              78           87              98
3       kareena                67              98           54              56
4       Mitansh                89              67           98              45

In [30]:
# First three rows in the data frame
report_card_df.head(3)

Student_names  chemistry grades  physics_grades  math_grades  biology_grades
0      Vanshika                23              23           56              45
1        Vivaan                45              56           76              78
2        Vrutik                45              78           87              98

In [31]:
# random 3 rows in the data frame
report_card_df.sample(3)

Student_names  chemistry grades  physics_grades  math_grades  biology_grades
0      Vanshika                23              23           56              45
2        Vrutik                45              78           87              98
3       kareena                67              98           54              56

In [32]:
# Describing all the numeric columns in the report card data frame
report_card_df.describe()

chemistry grades  physics_grades  math_grades  biology_grades
count          5.000000        5.000000     5.000000         5.00000
mean          53.800000       64.400000    74.200000        64.40000
std           25.083859       27.862161    19.188538        23.11493
min           23.000000       23.000000    54.000000        45.00000
25%           45.000000       56.000000    56.000000        45.00000
50%           45.000000       67.000000    76.000000        56.00000
75%           67.000000       78.000000    87.000000        78.00000
max           89.000000       98.000000    98.000000        98.00000

# Insights

- What is the average chemistry grade? 53.8
- What is the average biology grade? 64.4
- What is the lowest maths grade? 54
- What is the highest physics grade? 98
- How many scores are present in maths coulmn? 5

In [33]:
# Make a total grade column
report_card_df["total_grades"]=report_card_df["chemistry grades"]+report_card_df["physics_grades"]+report_card_df["math_grades"]+report_card_df["biology_grades"]
report_card_df

Student_names  chemistry grades  physics_grades  math_grades  \
0      Vanshika                23              23           56   
1        Vivaan                45              56           76   
2        Vrutik                45              78           87   
3       kareena                67              98           54   
4       Mitansh                89              67           98   

   biology_grades  total_grades  
0              45           147  
1              78           255  
2              98           308  
3              56           275  
4              45           299

In [34]:
# Make a column called average grade
report_card_df["average_grade"]=report_card_df["total_grades"]/4
report_card_df

Student_names  chemistry grades  physics_grades  math_grades  \
0      Vanshika                23              23           56   
1        Vivaan                45              56           76   
2        Vrutik                45              78           87   
3       kareena                67              98           54   
4       Mitansh                89              67           98   

   biology_grades  total_grades  average_grade  
0              45           147          36.75  
1              78           255          63.75  
2              98           308          77.00  
3              56           275          68.75  
4              45           299          74.75

In [35]:
report_card_df.describe()

chemistry grades  physics_grades  math_grades  biology_grades  \
count          5.000000        5.000000     5.000000         5.00000   
mean          53.800000       64.400000    74.200000        64.40000   
std           25.083859       27.862161    19.188538        23.11493   
min           23.000000       23.000000    54.000000        45.00000   
25%           45.000000       56.000000    56.000000        45.00000   
50%           45.000000       67.000000    76.000000        56.00000   
75%           67.000000       78.000000    87.000000        78.00000   
max           89.000000       98.000000    98.000000        98.00000   

       total_grades  average_grade  
count      5.000000        5.00000  
mean     256.800000       64.20000  
std       64.793518       16.19838  
min      147.000000       36.75000  
25%      255.000000       63.75000  
50%      275.000000       68.75000  
75%      299.000000       74.75000  
max      308.000000       77.00000

In [36]:
report_card_insights_df=report_card_df.describe().transpose()
report_card_insights_df

count   mean        std     min     25%     50%     75%  \
chemistry grades    5.0   53.8  25.083859   23.00   45.00   45.00   67.00   
physics_grades      5.0   64.4  27.862161   23.00   56.00   67.00   78.00   
math_grades         5.0   74.2  19.188538   54.00   56.00   76.00   87.00   
biology_grades      5.0   64.4  23.114930   45.00   45.00   56.00   78.00   
total_grades        5.0  256.8  64.793518  147.00  255.00  275.00  299.00   
average_grade       5.0   64.2  16.198380   36.75   63.75   68.75   74.75   

                    max  
chemistry grades   89.0  
physics_grades     98.0  
math_grades        98.0  
biology_grades     98.0  
total_grades      308.0  
average_grade      77.0

In [37]:
# Saving report card insights as csv file
report_card_insights_df.to_csv("Report Card Insights.csv")

In [41]:
report_card_df.loc[report_card_df["total_grades"]>= 140 ,"Status"]="pass"
report_card_df.loc[report_card_df["total_grades"]<140 , "Status"] ="fail"
report_card_df

Student_names  chemistry grades  physics_grades  math_grades  \
0      Vanshika                23              23           56   
1        Vivaan                45              56           76   
2        Vrutik                45              78           87   
3       kareena                67              98           54   
4       Mitansh                89              67           98   

   biology_grades  total_grades  average_grade Status  
0              45           147          36.75   pass  
1              78           255          63.75   pass  
2              98           308          77.00   pass  
3              56           275          68.75   pass  
4              45           299          74.75   pass

In [43]:
report_card_df.loc[(report_card_df["average_grade"]>= 90),"Character_grade"]="A"
report_card_df.loc[(report_card_df["average_grade"]>= 70) & (report_card_df["average_grade"]<90),"Character_grade"]= "B"
report_card_df.loc[(report_card_df["average_grade"]>= 50) & (report_card_df["average_grade"]<70),"Character_grade"]= "C"
report_card_df.loc[(report_card_df["average_grade"]>= 35) & (report_card_df["average_grade"]<50),"Character_grade"]= "D"
report_card_df.loc[(report_card_df["average_grade"]<35),"Character_grade"]="F"
report_card_df

Student_names  chemistry grades  physics_grades  math_grades  \
0      Vanshika                23              23           56   
1        Vivaan                45              56           76   
2        Vrutik                45              78           87   
3       kareena                67              98           54   
4       Mitansh                89              67           98   

   biology_grades  total_grades  average_grade Status Character_grade  
0              45           147          36.75   pass               D  
1              78           255          63.75   pass               C  
2              98           308          77.00   pass               B  
3              56           275          68.75   pass               C  
4              45           299          74.75   pass               B

In [44]:
report_card_df.to_csv("Report Card.csv")